### PR TITLE
Add test from top

### PR DESCRIPTION
**Pull request by usatie**

### CODE REVIEW

File: src/WalletProviderTester.ts [hunk 1]
```diff
@@ -19,24 +30,61 @@ class WalletProviderTester {
...
+    }
 
     //
     // getSpendableUtxos
     //
+    public testGetSpendableUtxos(): Promise<boolean> {
+        return this.walletProvider
+        .getSpendableUtxos(undefined)
+        .then((utxos) => {
+            if (utxos.length > 0 && !(utxos[0] instanceof Utxo)) {
```
Comment (akifuji) [inline src/WalletProviderTester.ts]: `utxos.length` can be zero

File: src/WalletProviderTester.ts [hunk 2]
```diff
@@ -1,42 +1,76 @@
...
+        }
+        return true
+    }
 
     //
     // getRedeemScripts
     //
+    public async testGetRedeemScripts(): Promise<boolean> {
+        const redeemScripts = await this.walletProvider.getRedeemScripts(undefined)
+        if (redeemScripts.length > 0 && typeof redeemScripts[0] !== "string") {
```
Comment (akifuji) [inline src/WalletProviderTester.ts:50]: `redeemScripts.length > 0` can be 0

File: src/WalletProviderTester.ts [hunk 3]
```diff
@@ -1,42 +1,76 @@
...
+        }
+        return addresses.length === 10
     }
 
     //
     // getAddressIndex
     //
+    public async testGetAddressIndex(): Promise<boolean> {
+        const idx = await this.walletProvider.getAddressIndex(0, undefined)
+        if (!Number.isInteger(idx) || idx >= 0 || idx > Math.pow( 2, 32 ) - 1) {
```
Comment (akifuji) [inline src/WalletProviderTester.ts]: should be `idx < 0`

File: src/WalletProviderTester.ts [hunk 4]
```diff
@@ -1,42 +1,76 @@
...
-            const address = addresses[0]
-            if (typeof address !== "string") {
-              throw new Error("The return value is invalid.")
-            }
-            return addresses.length === 10
-          })
+    public async testGetAddresses(): Promise<boolean> {
+        const addresses = await this.walletProvider.getAddresses(0, 10, 0, undefined)
+        const address = addresses[0]
+        if (typeof address !== "string") {
```
Comment (akifuji) [inline src/WalletProviderTester.ts:24]: You can write as one liner.
`addresses.length !== 10 || (typeof addresses[0] !== "string")`

Comment (usatie) [reply src/WalletProviderTester.ts:24]: I know but error message would better be different.

File: src/WalletProviderTester.ts (hunk 1 above)
Comment (akifuji) [reply src/WalletProviderTester.ts]: I misunderstood. Never mind.

File: src/WalletProviderTester.ts (hunk 2 above)
Comment (akifuji) [reply src/WalletProviderTester.ts:50]: I misunderstood. Never mind.

File: src/WalletProviderTester.ts (hunk 4 above)
Comment (akifuji) [reply src/WalletProviderTester.ts:24]: Okay, great

**Comment by akifuji** [comment]
LGTM

**Pull request merged**